File: Java/src/SampleDao.java
import java.io.BufferedReader;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.LinkedList;

public class SampleDao {

	public LinkedList<Sample> getSamples() {
			
		LinkedList<Sample> samples = new LinkedList<Sample>();
	
	    try(FileReader fr = new FileReader("samples.fft");
	        BufferedReader br = new BufferedReader(fr);) {
	
	      String line;
	
	      while((line = br.readLine()) != null) {
	
	        String [] rawSample = line.split(":");
	        String gesture = rawSample[0];
	
	        String [] rawFft = rawSample[1].split(",");
	        double [] fft = Utils.stringArrayToDoubleArray(rawFft);
	        
	        String [] rawJerkVector = rawSample[2].split(",");
	        double [] jerkVector = Utils.stringArrayToDoubleArray(rawJerkVector);
	
	        Sample sample = new Sample(fft, jerkVector, gesture);
	        samples.add(sample);
	        
	        System.out.println("Added new sample: " + sample.getGesture());
	      }
	    }catch(FileNotFoundException e){
	      System.out.println("Encountered a FileNotFoundException while reading fft file.");
	    }catch(IOException e) {
	      System.out.println("Encountered an IOException while reading fft file.");
	    }
	    
	    return samples;
	}
	
	public void writeSamples(double [] fftX, double [] fftY, double [] fftZ, double [] jerkVector) {
		
		try (PrintWriter writer = new PrintWriter("samples.fft");){
			String outputString = "";
					
			for(int i=0;i<fftX.length;i++) {
				outputString += fftX[i] + ",";
			}
			
			outputString += ":";
			
			for(int i=0;i<fftY.length;i++) {
				outputString += fftY[i] + ",";
			}
			
			outputString += ":";
			
			for(int i=0;i<fftZ.length;i++) {
				outputString += fftZ[i] + ",";
			}
			
			outputString += ":";
			
			for(int i=0;i<jerkVector.length;i++) {
				outputString += jerkVector[i];
			}
			writer.println(outputString);
			writer.flush();
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		}
	}
}

File: webInterface/gestureRecognition/sampledao.php
<?php

namespace GestureRecognition;

class SampleDao
{

    public function getSamples()
    {
        $samples = new SplDoublyLinkedList();

        try {
            $fr = new FileReader("samples.txt");
            $br = new BufferedReader($fr);
            $line;

            while (($line = $br.readLine()) != null) {
                $rawSample = $line.split(":");
                $gesture = $rawSample[0];

                $rawDataX = $rawSample[1].split(",");
                $dataX = Utils::stringArrayToDoubleArray($rawDataX);

                $rawDataY = $rawSample[2].split(",");
                $dataY = Utils::stringArrayToDoubleArray($rawDataY);

                $rawDataZ = $rawSample[3].split(",");
                $dataZ = Utils::stringArrayToDoubleArray($rawDataZ);

                $jerkVector = null;

                if (count($rawSample) == 5) {
                    $rawJerkVector = $rawSample[4].split(",");
                    $jerkVector = Utils::stringArrayToDoubleArray($rawJerkVector);
                }

                $sample = new Sample($dataX, $dataY, $dataZ, $jerkVector, $gesture);
                $samples.add($sample);

                echo("Added new sample: " + $sample.getGesture());
            }
        } catch (FileNotFoundException $e) {
            echo("Encountered a FileNotFoundException while reading fft file.");
        } catch (IOException $e) {
            echo("Encountered an IOException while reading fft file.");
        }

        return $samples;
    }
    
    public function writeSamples($gestureName, $dataX, $dataY, $dataZ, $jerkVector)
    {
        
        try {
            $writer = new PrintWriter(new FileOutputStream("samples.fft", true));
            $outputString = $gestureName;
                    
            for ($i=0; $i<count($dataX); $i++) {
                $outputString += $dataX[$i];

                if ($i < count($dataX)-1) {
                    $outputString += ",";
                } else {
                    $outputString += ":";
                }
            }
            
            for ($i=0; $i<count($dataY); $i++) {
                $outputString += $dataY[$i];

                if ($i < count($dataY)-1) {
                    $outputString += ",";
                } else {
                    $outputString += ":";
                }
            }
            
            for ($i=0; $i<count($dataZ); $i++) {
                $outputString += $dataZ[$i];

                if ($i < count($dataZ)-1) {
                    $outputString += ",";
                } else {
                    $outputString += ":";
                }
            }
            
            $outputString += ":";
            
            for ($i=0; $i<count($jerkVector); $i++) {
                $outputString += $jerkVector[$i] + ",";
            }

            $writer.println($outputString);
            $writer.flush();
        } catch (FileNotFoundException $e) {
            $e.printStackTrace();
        }
    }
}
